apm-agent-plugins/apm-jmx-plugin/src/main/java/co/elastic/apm/agent/jmx/JmxMetricTracker.java

```diff
@@ -175,19 +186,48 @@ synchronized void init(final MBeanServer platformMBeanServer) {
         jmxConfiguration.getCaptureJmxMetrics().addChangeListener(new ConfigurationOption.ChangeListener<List<JmxMetric>>() {
             @Override
             public void onChange(ConfigurationOption<?> configurationOption, List<JmxMetric> oldValue, List<JmxMetric> newValue) {
-                List<JmxMetricRegistration> oldRegistrations = compileJmxMetricRegistrations(oldValue, platformMBeanServer);
-                List<JmxMetricRegistration> newRegistrations = compileJmxMetricRegistrations(newValue, platformMBeanServer);
+                List<JmxMetric> registrationErrors = new ArrayList<JmxMetric>(); // those are not needed
+                List<JmxMetricRegistration> oldRegistrations = compileJmxMetricRegistrations(oldValue, platformMBeanServer, registrationErrors);
+
+                List<JmxMetricRegistration> newRegistrations;
+                synchronized (failedMetrics) {
+                    failedMetrics.clear();
+                    newRegistrations = compileJmxMetricRegistrations(newValue, platformMBeanServer, failedMetrics);
+                }
+
 
                 for (JmxMetricRegistration addedRegistration : removeAll(oldRegistrations, newRegistrations)) {
                     addedRegistration.register(platformMBeanServer, metricRegistry);
                 }
                 for (JmxMetricRegistration deletedRegistration : removeAll(newRegistrations, oldRegistrations)) {
                     deletedRegistration.unregister(metricRegistry);
                 }
-
             }
         });
-        register(jmxConfiguration.getCaptureJmxMetrics().get(), platformMBeanServer);
+
+        ConfigurationOption<TimeDuration> failedRetryConfig = jmxConfiguration.getFaildRetryInterval();
+        long retryMillis = failedRetryConfig.getValue().getMillis();
```

Comment:
scope is inside the if statement, why here?

Reply:
Fixed in 91f49cf3d